packages/blocks/shuffle/src/shuffle.tsx

```diff
@@ -130,49 +171,108 @@ export const Shuffle: BlockComponent<BlockEntityProperties> = ({
       }),
     );
 
+  const handleAddEntitiesClick = () => dialogRef.current?.show();
+
+  const handleRemoveAllClick = () => {
+    // we also want to remove all links for the linked items
+    const linkIds = draftItems
+      .map((item) => item.linkId)
+      .filter(Boolean) as string[];
+
+    void Promise.all(
+      linkIds.map((linkId) =>
+        graphService?.deleteLink({
+          data: { linkId },
+        }),
+      ),
+    );
+
+    updateItems([], true);
+  };
+
+  // adds the items selected on AddEntitiesDialog to the list
+  const handleAddEntityItems = (entityItems: Items) => {
+    updateItems(
+      produce(draftItems, (newItems) => {
+        return newItems.concat(entityItems);
+      }),
+      true,
+    );
+  };
+
+  /**
+   * linked items does not store a `value`,
+   * instead we use `entityId` to set the up-to-date entity label as `value`
+   * this way, we don't show a stale `value` if the linked entity gets updated
+   */
+  const enhancedDraftItems = useMemo(() => {
+    return draftItems.map((item) => {
+      const { entityId: itemEntityId } = item;
```

Comment:
```suggestion
      const itemEntityId = item.entityId;
```

Reply:
This reads better, nice suggestion!

Reply:
Updated here https://github.com/hashintel/hash/pull/1033/commits/36630dffdbf1e911bcea9df7150205dbdf5cc4e2